File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] result = new int[n];

        Stack<Integer> st = new Stack<>();
        for (int i = 0; i < n; i++) {
            while (!st.empty() && temperatures[i] > temperatures[st.peek()]) {
                int index = st.pop();
                result[index] = i - index;
            }
            st.push(i);

        }

        return result;

    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        stack = [0]
        result = [0] * n

        for i in range(1,n):
            while stack and temperatures[stack[-1]] < temperatures[i]:
                idx = stack.pop()
                result[idx] = i-idx
            stack.append(i)

        return result
